zilliqa/src/node.rs

```diff
@@ -495,12 +495,13 @@ impl Node {
             response.blocks.len()
         );
         let mut was_new = false;
+        let length_recvd = response.blocks.len();
 
         for block in response.blocks {
             was_new = self.consensus.receive_block(block)?;
         }
 
-        if was_new {
+        if was_new && length_recvd > 1 {
```

Comment:
Essentially, when you only get a single block back from a request that should be many blocks, you can assume there is nothing more to request.